File: Northern Frontier Logistics/claude1.py
import sys

def can_ship(W, cap, D):
    days, cur = 1, 0
    for w in W:
        if cur + w > cap:
            days += 1
            cur = 0
        cur += w
    return days <= D

def min_capacity(W, D):
    lo, hi = max(W), sum(W)
    while lo < hi:
        mid = (lo + hi) // 2
        if can_ship(W, mid, D):
            hi = mid
        else:
            lo = mid + 1
    return lo

def main():
    data = sys.stdin.read().split()
    idx = 0
    T = int(data[idx]); idx += 1
    results = []
    for _ in range(T):
        N, D = int(data[idx]), int(data[idx+1])
        idx += 2
        W = list(map(int, data[idx:idx+N]))
        idx += N
        results.append(str(min_capacity(W, D)))
    sys.stdout.write("\n".join(results) + "\n")

if __name__ == "__main__":
    main()

File: Northern Frontier Logistics/claude1.cpp
#include <bits/stdc++.h>
using namespace std;

bool canShip(vector<int>& W, int cap, int D){
    int days = 1, cur = 0;
    for(int w : W){
        if(cur + w > cap){
            days++;
            cur = 0;
        }
        cur += w;
    }
    return days <= D;
}

int main(){
    int T;
    cin >> T;
    while(T--){
        int N, D;
        cin >> N >> D;
        vector<int> W(N);
        int lo = 0, hi = 0;
        for(auto &w : W){
            cin >> w;
            lo = max(lo, w);
            hi += w;
        }
        while(lo < hi){
            int mid = lo + (hi - lo) / 2;
            if(canShip(W, mid, D)) hi = mid;
            else lo = mid + 1;
        }
        cout << lo << "\n";
    }
}